client/src/app/pages/applications/applications-table/applications-table.tsx

```diff
@@ -826,16 +833,31 @@ export const ApplicationsTable: React.FC = () => {
           <Thead>
             <Tr>
               <TableHeaderContentWithControls {...tableControls}>
-                <Th {...getThProps({ columnKey: "name" })} width={15} />
-                <Th
-                  {...getThProps({ columnKey: "businessService" })}
-                  width={15}
-                />
-                <Th {...getThProps({ columnKey: "assessment" })} width={10} />
-                <Th {...getThProps({ columnKey: "review" })} width={10} />
-                <Th {...getThProps({ columnKey: "analysis" })} width={10} />
-                <Th {...getThProps({ columnKey: "tags" })} width={10} />
-                <Th {...getThProps({ columnKey: "effort" })} width={10} />
+                {/* {columnState.columns. isVisible then render this column} */}
+                {getColumnVisibility("name") && (
```

Comment:
I don't mind this approach.

The `visibility` prop exists and is used in the [PF docs](https://www.patternfly.org/components/table#cell-width-breakpoint-modifiers) to use the css breakpoints to show/hide columns.  Fun fact is that prop is not in any of the generated docs and it needs to be tracked down in sources.  I tried that and got lost pretty quick.  Probably isn't worth trying to use `visibility` for the user based show/hide column support.